packages/atomic/src/components/commerce/product-template/product-template-common.tsx

```diff
@@ -0,0 +1,185 @@
+    const hasValidParent = validParents
+      .map((p) => p.toUpperCase())
+      .includes(host.parentElement?.nodeName || '');
+    const tagName = host.nodeName.toLowerCase();
+
+    if (!hasValidParent) {
+      setError(
+        new Error(
+          `The "${tagName}" component has to be the child of one of the following: ${validParents
+            .map((p) => `"${p.toLowerCase()}"`)
+            .join(', ')}.`
+        )
+      );
+      return;
+    }
+
+    const template = host.querySelector('template');
+    if (!template) {
+      setError(
+        new Error(
+          `The "${tagName}" component has to contain a "template" element as a child.`
+        )
+      );
+      return;
+    }
+
+    if (!allowEmpty && !template.innerHTML.trim()) {
+      setError(
+        new Error(`The "template" tag inside "${tagName}" cannot be empty.`)
+      );
+      return;
+    }
+
+    if (template.content.querySelector('script')) {
+      console.warn(
+        'Any "script" tags defined inside of "template" elements are not supported and will not be executed when the products are rendered.',
+        host
+      );
+    }
+
+    const {section: sectionNodes, other: otherNodes} = groupNodesByType(
+      template.content.childNodes
+    );
+    if (sectionNodes?.length && otherNodes?.length) {
+      console.warn(
+        'Product templates should only contain section elements or non-section elements. Future updates could unpredictably affect this product template.',
+        host,
+        {sectionNodes, otherNodes}
+      );
+    }
+  }
+
+  getTemplate(
+    conditions: ProductTemplateCondition[],
+    error: Error
+  ): ProductTemplate<TemplateContent> | null {
+    if (error) {
+      return null;
+    }
+
+    return {
+      conditions: conditions.concat(this.matchConditions),
+      content: getTemplateElement(this.host).content!,
+      priority: 1,
+    };
+  }
+
+  renderIfError(error: Error) {
+    if (error) {
+      return (
+        <atomic-component-error
+          element={this.host}
+          error={error}
+        ></atomic-component-error>
+      );
+    }
+  }
+}
+
+function getTemplateElement(host: HTMLElement) {
+  return host.querySelector('template')!;
+}
+
+// TODO: implement match conditions for product templates
+/* export function makeMatchConditions(
+  mustMatch: Record<string, string[]>,
+  mustNotMatch: Record<string, string[]>
+): ProductTemplateCondition[] {
+  const conditions: ProductTemplateCondition[] = [];
+  for (const field in mustMatch) {
+    conditions.push(
+      ResultTemplatesHelpers.fieldMustMatch(field, mustMatch[field])
+    );
+  }
+
+  for (const field in mustNotMatch) {
+    conditions.push(
+      ResultTemplatesHelpers.fieldMustNotMatch(field, mustNotMatch[field])
+    );
+  }
+  return conditions;
+} */
+
+// TODO: Use ProductTemplatesHelpers when implemented
+export function makeDefinedConditions(
+  ifDefined?: string,
+  ifNotDefined?: string
+): ResultTemplateCondition[] {
```

Comment:
Since those are not currently ready, can we make them no-op functions, and remove the imports on `ResultTemplateCondition, ResultTemplatesHelpers` from `@coveo/headless` ?